homeassistant/components/mqtt/mixins.py

```diff
@@ -1120,7 +1121,22 @@ def _setup_common_attributes_from_config(self, config: ConfigType) -> None:
             config.get(CONF_ENABLED_BY_DEFAULT)
         )
         self._attr_icon = config.get(CONF_ICON)
-        self._attr_name = config.get(CONF_NAME)
+        entity_name: str | None = config.get(CONF_NAME)
+        if CONF_DEVICE in config:
+            device_name: str | None = config[CONF_DEVICE].get(CONF_NAME)
```

Comment:
Instead of accessing the config dictionary here, should we add some helper or feature to `MqttEntityDeviceInfo`? Maybe `MqttEntityDeviceInfo.has_device_name` or `MqttEntityDeviceInfo.has_device`?